src/Clock.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+type State = {
+  date: Date,
+};
+
+type Props = {
+  clockName: string,
+};
+
+export class Clock extends React.Component<Props, State> {
+  state = {
+    date: new Date(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.tick();
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const oldName = prevProps.clockName;
+    const newName = this.props.clockName;
+
+    if (oldName !== newName) {
+      const message = `Renamed from ${oldName} to ${newName}`;
+
+      // eslint-disable-next-line no-console
+      console.log(message);
+    }
+  }
+
+  componentWillUnmount() {
+    window.clearInterval(this.timerId);
+  }
+
+  tick() {
```

Comment:
Method's name should start with a verb (so you could easily tell what your method actually do)